README.md

```diff
@@ -142,6 +154,25 @@ config.extensionFunctionCreator = (functionName: RDF.NamedNode) => {
 }
 ```
 
+### Overload function caching
+
+An overloadcache allows Sparqlee to cache the implementation of a function provided the argument types. 
+This cache is only used when provided to the context.
+
+### Super type discovery
+
+The `superTypeDiscoverCallback` allow a user to use custom types and define there super relationship to other types.
```

Comment:
```suggestion
The `superTypeDiscoverCallback` allow a user to use custom types and define their super relationship to other types.
```

Reply:
Can you also explain why this is needed/useful?

Reply:
Sure! Because it's cool :)